lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -798,9 +798,28 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     std::string Result = Mangler.mangleTypeForDebugger(
         Ty, nullptr);
 
+    // Returns true iff the given type comes from an implicit declaration
+    // generated by the differentiation transform: a linear map struct or
+    // branching trace enum.
+    auto isAutoDiffGeneratedDeclarationType = [](Type type) -> bool {
```

Comment:
Could this also check that differentiation is enabled? This would protect more against accidentally returning true on some type that was generated by something non-AD.

Reply:
Sounds good! I'll add this in a follow-up to minimize CI invocations.